cmd/cluster-node-tuning-operator/main.go

```diff
@@ -1,64 +1,149 @@
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
 	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
 
-	tunedv1 "github.com/openshift/cluster-node-tuning-operator/pkg/apis/tuned/v1"
+	v1 "github.com/openshift/cluster-node-tuning-operator/pkg/apis/tuned/v1"
+	"github.com/openshift/cluster-node-tuning-operator/pkg/config"
 	"github.com/openshift/cluster-node-tuning-operator/pkg/metrics"
 	"github.com/openshift/cluster-node-tuning-operator/pkg/operator"
 	"github.com/openshift/cluster-node-tuning-operator/pkg/signals"
 	"github.com/openshift/cluster-node-tuning-operator/pkg/tuned"
+	"github.com/openshift/cluster-node-tuning-operator/pkg/util"
 	"github.com/openshift/cluster-node-tuning-operator/version"
 )
 
+const (
+	operandFilename  = "openshift-tuned"
+	operatorFilename = "cluster-node-tuning-operator"
+	metricsHost      = "0.0.0.0"
+	leaderElectionID = "node-tuning-operator-lock"
+)
+
 var (
-	boolVersion = flag.Bool("version", false, "show program version and exit")
-	boolLocal   = flag.Bool("local", false, "local run outside a pod")
+	scheme = apiruntime.NewScheme()
 )
 
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(v1.AddToScheme(scheme))
+	utilruntime.Must(mcov1.AddToScheme(scheme))
+	utilruntime.Must(apiconfigv1.Install(scheme))
+}
+
 func printVersion() {
 	klog.Infof("Go Version: %s", runtime.Version())
 	klog.Infof("Go OS/Arch: %s/%s", runtime.GOOS, runtime.GOARCH)
-	klog.Infof("%s Version: %s", tunedv1.TunedClusterOperatorResourceName, version.Version)
+	klog.Infof("%s Version: %s", v1.TunedClusterOperatorResourceName, version.Version)
 }
 
 func main() {
-	const (
-		operandFilename  string = "openshift-tuned"
-		operatorFilename string = "cluster-node-tuning-operator"
-	)
+	var boolVersion bool
+	var enableLeaderElection bool
+	flag.BoolVar(&boolVersion, "version", false, "show program version and exit")
+	flag.BoolVar(&enableLeaderElection, "enable-leader-election", true,
+		"Enable leader election for controller manager. "+
+			"Enabling this will ensure there is only one active controller manager.")
 
 	runAs := filepath.Base(os.Args[0])
 
-	stopCh := signals.SetupSignalHandler()
-
 	switch runAs {
 	case operatorFilename:
+		var metricsAddr string
+		flag.StringVar(&metricsAddr, "metrics-addr", fmt.Sprintf("%s:%d", metricsHost, metrics.MetricsPort),
+			"The address the metric endpoint binds to.")
+
 		klog.InitFlags(nil)
 		flag.Parse()
 
 		printVersion()
 
-		if *boolVersion {
+		if boolVersion {
 			os.Exit(0)
 		}
 
-		go metrics.RunServer(metrics.MetricsPort, stopCh)
-		metrics.RegisterVersion(version.Version)
-		controller, err := operator.NewController()
+		// we have two namespaces that we need to watch
+		// 1. NTO namespace - for NTO resources
+		// 2. None namespace - for cluster wide resources
+		ntoNamespace := config.OperatorNamespace()
+		namespaces := []string{
+			ntoNamespace,
+			metav1.NamespaceNone,
```

Comment:
I understand that its because  `MultiNamespacedCacheBuilder` by default creates a global cache for cluster scoped resource. But we also have a need to watch pods in all namespaces when tuned CRs are pod typed driven. 
In that case would you also recommend to remove the `metav1.NamespaceNone` from the cache builder ?

Reply:
mm good point, let leave it for now